File: 01-solution.py
solution = {'a': 0, 'b': 0}

with open('01-input.txt', 'r') as f:
    elves = f.read().strip().split('\n\n')

elves = [elf.strip().split('\n') for elf in elves]
elves = [sum([int(x) for x in elf]) for elf in elves]

elves.sort()
solution['a'] = elves[-1]
solution['b'] = sum(elves[-3:])

print(solution)

File: 01-solution.js
const { readFile } = require('fs/promises')

async function solve () {

  const file = await readFile('./01-input.txt')
  const elves = file
    .toString()
    .trim()
    .split('\n\n')
    .map(elf => elf.split('\n'))
    .map(elf => arraySum(elf))
    .sort()

  console.log(elves.at(-1))
  console.log(arraySum(elves.slice(-3)))

}

const arraySum = arr => arr.reduce((s, x) => s + Number(x), 0)

solve()
